src/context/mod.rs

```diff
@@ -229,6 +229,13 @@ pub trait Context<'a>: ContextInternal<'a> {
         result
     }
 
+    #[cfg(feature = "try-catch-api")]
+    fn try_catch<'b: 'a, F>(&mut self, f: F) -> Result<Handle<'a, JsValue>, Handle<'a, JsValue>>
```

Comment:
The current RFC defines the `Ok` value as a generic `T: Value`. This is feature flagged, so this is pretty minor.

Reply:
Yeah, as I was hacking I had myself convinced that didn't make sense and the RFC was wrong. But now that I think about it, I guess it's just strictly more expressive, since choosing a given `T` gives you an implicit downcast, and if you want the dynamic version you can always choose `JsValue` as your `T`.

Reply:
Oh! I just figured out why it's not clear how to make this generic in `T: Value`. It means we have to do a downcast, which can throw a `TypeError`. But we're trying to catch exceptions in this API. And we don't even return a `NeonResult`. I think I'm going to leave the API as-is for now, since it's feature-flagged, and we can discuss the finer points of this in the RFC.

Reply:
Follow-up to the follow-up… @kjvalencik pointed out on Slack that the way to make it generic is for the type signature to enforce that the callback returns a `T` as well. Once I understood this it was pretty easy to implement!